float32 height
float32 tcenter_x
float32 tcenter_y
float32 tcenter_z
float32 ttop_x
float32 ttop_y
float32 ttop_z